setup.py

```diff
@@ -23,40 +24,66 @@ def get_tag(self):
         return python, abi, plat
 
 
-def download_so():
-    uname = platform.uname()
-    machine = uname.machine
+def abs_machine():
+    machine = platform.machine()
+
+    pointer_bits = struct.calcsize("P") * 8
+    if pointer_bits not in (32, 64):
+        raise Exception("Unsupported pointer size")
+
+    is_64 = pointer_bits == 64
+
+    # x86 based archs
+    if machine in ('AMD64', 'x86_64', 'i686-64', 'i386', 'i686', 'x86'):
```

Comment:
Where does these machine types come from?

Reply:
I checked the possible `uname`s list on [wikipedia](https://en.wikipedia.org/wiki/Uname), `'i686-64', 'i386', 'x86'` seemed to be rare or outdated.

Reply:
> Where does these machine types come from?

https://en.wikipedia.org/wiki/Uname (from `Machine -m Posix` column)


> I checked the possible `uname`s list on [wikipedia](https://en.wikipedia.org/wiki/Uname), `'i686-64', 'i386', 'x86'` seemed to be rare or outdated.

Thanks, updated.